app/controllers/UploadSupportingDocumentsController.scala

```diff
@@ -46,16 +55,43 @@ class UploadSupportingDocumentsController @Inject() (
   ): Action[AnyContent] =
     (identify andThen getData(draftId) andThen requireData).async {
       implicit request =>
-        val answers    = request.userAnswers
-        val fileStatus = answers.get(UploadSupportingDocumentPage)
-
-        helper.onPageLoadWithFileStatus(
-          mode,
-          draftId,
-          errorCode,
-          key,
-          fileStatus,
-          isLetterOfAuthority = false
-        )
+        val numberOfAttachments = AllDocuments.get().getOrElse(List.empty).length
+
+        if (appConfig.agentOnBehalfOfTrader) {
+          if (
+            numberOfAttachments >= userRoleProvider
+              .getUserRole(request.userAnswers)
+              .getMaxFileUploads()
+          ) {
+            Future.successful(
+              Redirect(UploadAnotherSupportingDocumentController.onPageLoad(NormalMode, draftId))
+            )
```

Comment:
UploadSupportingDocumentsControllerSpec will need a test added for this redirect case.